cmd/kubeadm/app/master/clusterroles.go

```diff
@@ -0,0 +1,49 @@
+package master
+
+import (
+	"fmt"
+
+	api "k8s.io/kubernetes/pkg/api/v1"
+	rbac "k8s.io/kubernetes/pkg/apis/rbac/v1alpha1"
+	"k8s.io/kubernetes/pkg/client/clientset_generated/clientset"
+)
+
+// CreateBootstrapRBACClusterRole creates the necessary ClusterRole for bootstrapping
+func CreateBootstrapRBACClusterRole(clientset *clientset.Clientset) error {
+	clusterRole := rbac.ClusterRole{
+		// a role to use for setting up a proxy
+		ObjectMeta: api.ObjectMeta{Name: "system:kubelet-bootstrap"},
+		Rules: []rbac.PolicyRule{
+			// Used to build serviceLister
+			rbac.NewRule("get").Groups("").Resources("nodes").RuleOrDie(),
```

Comment:
Correct. I'd even like to see the auto-CSR-approval change from a flag to a permission check that could be dynamically granted/revoked. /cc @mikedanese